Expensify/App
# [WIP] Create local value state, debounce onChangeText

cc @marcaaron 

### Details

We were already storing a local copy of the value in `this.value` (outside of `this.state`), not sure why 🤷 

Removed `this.value` and added `this.state.localValue` in `BaseExpensiTextInput`
On every change we update and read the input value from `this.state.localValue`, and we debounce calling the callback `onChangeText`.

Delay when typing decreased considerably.

https://user-images.githubusercontent.com/87341702/132926146-d265bd8f-3bce-4a3d-a70c-d375cc30f892.mov


### Fixed Issues

$ https://github.com/Expensify/App/issues/5209

### Tests and QA

1. Navigate to `/bank-account/company`
2. Type in a text input, should behave normally (text appearing)
3. Click "Save & continue"
4. Close the sidebar
5. Navigate again to  `/bank-account/company`
6. Values entered before should be there

Other tests: Input values that cause inline validation errors (red outline, error text), typing in the fields with error should make the errors disappear with some small delay. 

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
